target/linux/ath79/mikrotik/base-files/etc/board.d/01_leds

```diff
@@ -7,6 +7,13 @@ board_config_update
 board=$(board_name)
 
 case "$board" in
+mikrotik,routerboard-2011)
+	ucidef_set_led_switch "eth6" "ETH6" "green:eth6" "switch1" "0x20"
+	ucidef_set_led_switch "eth7" "ETH7" "green:eth7" "switch1" "0x10"
+	ucidef_set_led_switch "eth8" "ETH8" "green:eth8" "switch1" "0x08"
+	ucidef_set_led_switch "eth9" "ETH9" "green:eth9" "switch1" "0x04"
+	ucidef_set_led_switch "eth10" "ETH10" "green:eth10" "switch1" "0x02"
```

Comment:
This part I believe is not yet working. But I have to admit I also do not fully understand how exactly this is supposed to work.

Reply:
This is direct c/p from 19.07 release. If I understood the UCI code correctly, it expects the LEDs with the specified names to be defined in the device tree, but I don't know how to define LEDs connected to pinmux.

Reply:
@adschm @robimarko 
Do you guys have a clue here?

Thanks

Reply:
I started looking into it but don't have anything concluding yet. The closest upstream thing I found so far is the following:
https://lwn.net/Articles/830947/
I guess we would need something similar for the ar8327 (resp. I still dream about having this all using DSA as well :smirk:).

Reply:
Unfortunately, no idea how is this supposed to work.
AR8327 switch should be able to control the LEDs on its own if configured to do so, so what's the purpose of these userspace triggers?

Reply:
@robimarko On RB2011 all 10 LEDs are wired to AR8327 switch. First five ports are GE connected to AR8327, the other five are connected to AR9344 internal FE switch. AR8327 is connected via GMAC0 in RGMII mode.
In 19.07, we have this in mach file:
```
static const struct ar8327_led_info rb2011_ar8327_leds[] = {
	AR8327_LED_INFO(PHY0_0, HW, "rb:green:eth1"),
	AR8327_LED_INFO(PHY1_0, HW, "rb:green:eth2"),
	AR8327_LED_INFO(PHY2_0, HW, "rb:green:eth3"),
	AR8327_LED_INFO(PHY3_0, HW, "rb:green:eth4"),
	AR8327_LED_INFO(PHY4_0, HW, "rb:green:eth5"),
	AR8327_LED_INFO(PHY0_1, SW, "rb:green:eth6"),
	AR8327_LED_INFO(PHY1_1, SW, "rb:green:eth7"),
	AR8327_LED_INFO(PHY2_1, SW, "rb:green:eth8"),
	AR8327_LED_INFO(PHY3_1, SW, "rb:green:eth9"),
	AR8327_LED_INFO(PHY4_1, SW, "rb:green:eth10"),
	AR8327_LED_INFO(PHY4_2, SW, "rb:green:usr"),
};
```

As well as the configuration in `01_leds`:
```
ucidef_set_led_switch "eth6" "ETH6" "rb:green:eth6" "switch1" "0x20"
ucidef_set_led_switch "eth7" "ETH7" "rb:green:eth7" "switch1" "0x10"
ucidef_set_led_switch "eth8" "ETH8" "rb:green:eth8" "switch1" "0x08"
ucidef_set_led_switch "eth9" "ETH9" "rb:green:eth9" "switch1" "0x04"
ucidef_set_led_switch "eth10" "ETH10" "rb:green:eth10" "switch1" "0x02"
```

My guess is that `swconfig` is instructed through this configuration to toggle the LED on AR8327 (because the AR8327 switch itself probably can't know on its own that something is plugged to AR9344), but I don't know how to port this configuration to the device tree.

(P.S. First five LEDs are not defined anywhere.)

Reply:
Ok, then it makes sense as it's supposed to use the swconfig LED code to trigger the LEDs.
Those first 5 LEDs are HW triggered by the PHYs themself.